src/lib/Movement/moveTo.ts

```diff
@@ -92,6 +92,77 @@ export const moveTo = (
     clearCachedPath(creep, cache);
   }
 
+  // Dynamic choose weight for roads, plains and swamps depending on body.
```

Comment:
This should be invoked in `lib/Movement/generatePath.ts` instead, as we only need to calculate this when we're actually generating a path. Here, it will run each tick when the creep moves.